rust/src/index/vector/graph/builder.rs

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 Lance Developers.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Graph in memory.
+
+use super::Vertex;
+
+/// A graph node to hold the vertex data and its neighbors.
+#[derive(Debug)]
+pub(crate) struct Node<V: Vertex> {
+    pub(crate) vertex: V,
+    pub(crate) neighbors: Vec<u32>,
```

Comment:
are the neighbors other Node's? other Vertex's? or the row id? if row id - should it be u64?

Reply:
This neighour id / vertex id will be the location in the graph index file. which will be different order from "row id" that points to the original vector in the dataset.

`u32` can support up to 4B vectors per index. Feel that beyond that point, we need to apply IVF first? Main consideration is that  it reduces the I/O and memory footprints by half.